packages/core/vue/src/useLocalStorage.ts

```diff
@@ -0,0 +1,19 @@
+import { customRef, Ref } from '@vue/reactivity';
+
+export function useLocalStorage<T> (key: string, defaultValue: T | null = null): Ref<T | null> {
+    return customRef<T | null>((track, trigger) => ({
+        get: () => {
+            track();
+            const value = localStorage.getItem(key);
+            return value ? JSON.parse(value) : defaultValue;
```

Comment:
We don't do this yet in the React design, but we should -- wrap `JSON.parse` in try/catch and return the default value if it fails. This way users won't get an uncaught runtime error.